Fix append table

Scmadar opened the pull request:
As described in issues #561 and #563, the append_table function has been crashing in version 2.0.6. This is due to a change made for #546, to give the proper return value for this function (version 2.0.5 had no return value for append_table).

Instead of creating a GridRange using the JSON response text directly, e.g. "'Some Sheet'!A1:B4", split it apart and use the cell range only, e.g. "A1:B4". You don't need to add the worksheet name, because it's already being provided in self - and self comes with all sorts of goodies like the worksheet ID built in.

This appears to work correctly - the GridRange(s) created correctly link the worksheet, have the right worksheet ID and link to the overarching spreadsheet, etc. Confirmed it is working in a project, updating the range successfully and not throwing errors.

More testing may be needed in case the JSON response only returns the cell value, but I think Google is pretty good about sending the entire unambiguous range (including the sheet name).

Scmadar committed:
Update creation of return value for append_table

Scmadar committed:
Use rsplit instead, in case a sheet name has an !

If a sheet name has an exclamation point in it, split() will happily return the wrong range. rsplit() should split on the last !, which will be between the sheet name and the cell range. Still assumes that the range provided includes an !.

Scmadar commented:
My bad, does not fix #563 - that is a different error. Should resolve #561 though.

Scmadar committed:
Fix append_table return value if keys missing

The tableRange portion of the JSON response will not be present if the worksheet you are using is blank (or, I assume, if it can't detect a 'table' in the area you specify). To mimic the JSON response, the append_table method should also not include the tableRange.

Scmadar commented:
Added a separate fix for #563 - check for the key before adding that portion to the response.

Optionally, could still add the tableRange key with a null value. However, I would imagine that the returned value should mimic the Google response as closely as possible.

nithinmurali commented:
Thank you for the PR. Looks good to me. 
However since this is a bugfix, could you please add a test here https://github.com/nithinmurali/pygsheets/blob/staging/tests/online_test.py#L261 to confirm that it indeed is working.

Scmadar committed:
Add test_append_table()

Scmadar commented:
Added test for append_table. All 69 (nice) tests passing.
Thanks!

Scmadar committed:
Remove Python 3.6 from tests

Python 3.6 deprecated on ubuntu-latest - https://github.com/actions/setup-python/issues/544#issuecomment-1332535877

If v3.6 support is still needed to test against, add another job for it running on ubuntu-20.04

Scmadar committed:
Update test.yml

Update Github Actions versions - checkout@v3, setup-python@v4

nithinmurali commented:
Awesome. looks Good!
